app/controllers/concerns/foreman_rh_cloud/registration_manager_extensions.rb

```diff
@@ -0,0 +1,39 @@
+module ForemanRhCloud
+  module RegistrationManagerExtensions
+    include ForemanRhCloud::CertAuth
+
+    def logger
+      Rails.logger
+    end
+
+    def unregister_host(host, options = {})
+      organization_destroy = options.fetch(:organization_destroy, false)
+
+      # Reload to ensure we have fresh association data
+      host.reload
+
+      # Only delete from HBI in IoP mode (hosted mode uses async job for cleanup)
+      hbi_host_destroy(host) if ForemanRhCloud.with_iop_smart_proxy? && !organization_destroy && host.insights_facet&.uuid&.presence
```

Comment:
Re: `organization_destroy` - This line is modeled on the `candlepin_consumer_destroy` line in Katello's registration_manager. Katello cleans up Candlepin consumers in bulk during organization destroy, but foreman_rh_cloud currently doesn't have any corresponding cleanup attached to destroying organizations. The consequence would be that if you destroy an org, orphaned hosts will remain in HBI. (Or, if we removed the check here, possibly thousands of API calls to HBI during org destroy.) I'm thinking we should maybe add bulk HBI cleanup in a followup. Thoughts @parthaa @ShimShtein ?